pkg/inputs/snmp/snmp.go

```diff
@@ -371,6 +385,10 @@ func parseConfig(ctx context.Context, file string, log logger.ContextL) (*kt.Snm
 
 	// If there's a global user tags and match, add them in here.
 	if ms.Global != nil {
+		for p, m := range ms.Global.ProviderMap {
```

Comment:
I might just be missing it, but is `ProviderMap` always initialized?

Reply:
Its not a pointer so its always (implicitly) set up. I also went around on this for a while but I think this is the state.